apps/authentication-service/src/shared/http/httpService.ts

```diff
@@ -0,0 +1,142 @@
+import { ResponseError } from "auth0";
+import fetchIntercept, { FetchInterceptorResponse } from "fetch-intercept";
+
+import { AuthService, authEndpoint } from "@/shared/auth/authService";
+import { IAuthService } from "@/shared/auth/authModel";
+import {
+  IHttpDataRequest,
+  IHttpRefreshCallback,
+  IHttpRequestOptions,
+  IHttpService,
+} from "@/shared/http/httpModel";
+import { ISingleton } from "@/shared/models";
+
+const getRelativeUrl = (url: string): string => {
+  const _url = new URL(url);
+
+  return url.replace(_url.origin, "");
+};
+
+export const parseResponse = async (response: Response) => {
+  if (!response.ok) {
+    throw Error(response.statusText);
+  }
+  return response.json();
+};
+
+export const HttpService: ISingleton<IHttpService> = (() => {
```

Comment:
I have the same concerns here about creating a function that immediately calls itself, and the same concerns about why does this need to be instantiated

Reply:
Service removed.